enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -730,7 +730,7 @@ def remaining_balance_per_user(self, lms_user_id=None):
         Returns the remaining redeemable credit for the user.
         """
         existing_transaction_count = len(self.transactions_for_learner(lms_user_id)['transactions'])
-        return self.per_learner_enrollment_limit - existing_transaction_count
+        return int(self.per_learner_enrollment_limit or 0) - existing_transaction_count
```

Comment:
In the case where `per_learner_enrollment_limit` is `None` or the limit itself is `0`, I'm wondering if it conceptually makes sense for this method to return a negative number?

A negative number representing "remaining balance" feels a bit odd, when as implemented, the field really becomes the negative of "how many transactions has this learner has" which doesn't really make any semantic sense as "remaining balance", IMHO.

```python
>>> existing_transaction_count = 10
>>> per_learner_enrollment_limit = None
>>> (per_learner_enrollment_limit or 0) - existing_transaction_count
-10
```

I'm wondering if these methods should instead fail early and `return None` if there is no `self.per_learner_enrollment_limit` or `self.per_learner_spend_limit` set for the budget/policy?

Rationale: when there is no per-learner limit set for the budget, there can't really be a notion of a remaining balance specific to the learner, where it'd instead fallback to looking at the overall budget's remaining balance (not per-learner).

Example:

```python
def remaining_balance_per_user:
  if self.per_learner_enrollment_limit is None:
    return None

  if self.per_learner_enrollment_limit <= 0:
    return 0

  existing_transaction_count = len(self.transactions_for_learner(lms_user_id)['transactions'])
  return self.per_learner_enrollment_limit - existing_transaction_count
```

What do you think?

Reply:
That does make a lot of sense. I have changed the logic according to your suggestion.